### PR TITLE
Remove layers that don't belong in RandAugment from RandAugment

### PR DESCRIPTION
**Pull request by LukeWood**
Some of these are mean't to be applied BEFORE rand augment - CutOut, and some of these are just too strong (Invert).

This PR is a result of a discussion with the original RandAugment author.

Updated image:
randaug

### CODE REVIEW

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 1]
```diff
@@ -76,7 +79,7 @@ def __init__(
         augmentations_per_image=3,
         magnitude=0.5,
         magnitude_stddev=0.15,
-        rate=1.0,
+        rate=10 / 11,
```
Comment (kartik4949) [inline keras_cv/layers/preprocessing/rand_augment.py:82]: 1/11 or 10/11?

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:82]: 10/11 because this is the rate we apply augmentations at.

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:82]: Ha - no worries now I see you were looking at the docstring.  I fixed it - thanks Kartik!

Comment (kartik4949) [reply keras_cv/layers/preprocessing/rand_augment.py:82]: got confused from doctoring, cool

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 2]
```diff
@@ -58,7 +58,10 @@ class RandAugment(RandomAugmentationPipeline):
             Defaults to `0.15`.
         rate:  the rate at which to apply each augmentation.  This parameter is applied
             on a per-distortion layer, per image.  Should be in the range `[0, 1]`.
-            Defaults to `1`.
+            Defaults to `10/11`, as the original `RandAugment` paper includes an
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:61]: Not 100% sure we should do it. This adds overhead, and the actual benefits are unclear. We could simply leave it to 1.0 and mention that 10 / 11 reproduces the implementation proposed by the paper.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Changes LGTM except the `rate` default value.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py]: Done - yeah, this is kind of weird and you are right that it is unclear if it actually even helps.

Comment (bhack) [reply keras_cv/layers/preprocessing/rand_augment.py]: The ablation study was done only for  Wide-ResNet-28-2 models weretrained on CIFAR-10 using RandAugment (N = 3, M = 4)

immagine

Do we need to remove negative avg augmentations from the list? Do we know if this was confirmed for the other datasets/models?

Comment (bhack) [reply keras_cv/layers/preprocessing/rand_augment.py]: As always.. too late :)

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py]: > The ablation study was done only for `Wide-ResNet-28-2` models trained on `CIFAR-10` using `RandAugment (N = 3, M = 4)`
> 
> immagine
> 
> Do we need to remove negative avg augmentations from the list? Do we know if this was confirmed for the other datasets/models?

Hey @bhack - it is not confirmed for other datasets.  I discussed this with the original paper author.  He highly recommended the set of augmentations we now have.

Comment (bhack) [reply keras_cv/layers/preprocessing/rand_augment.py]: Ok thanks

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py]: > Ok thanks

I was very confused by the presence of negative values in the table in the paper too.  It is very confusing with respect to the point they're trying to show (that this is a good general purpose augmentation technique)

Comment (bhack) [reply keras_cv/layers/preprocessing/rand_augment.py]: > I was very confused by the presence of negative values in the table in the paper too. It is very confusing with respect to the point they're trying to show (that this is a good general purpose augmentation technique)

Yes I suppose that also a more trivial augmentation policy like TA could perform better in some cases:

immagine

**Pull request merged**